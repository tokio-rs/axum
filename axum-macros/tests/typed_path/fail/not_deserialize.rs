use axum_macros::TypedPath;

#[derive(TypedPath)]
#[typed_path("/users/:id")]
struct MyPath {
    id: u32,
}

fn main() {}
