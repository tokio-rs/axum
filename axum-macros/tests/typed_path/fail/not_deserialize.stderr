error[E0277]: the trait bound `for<'de> MyPath: serde::de::Deserialize<'de>` is not satisfied
 --> tests/typed_path/fail/not_deserialize.rs:3:10
  |
3 | #[derive(TypedPath)]
  |          ^^^^^^^^^ the trait `for<'de> serde::de::Deserialize<'de>` is not implemented for `MyPath`
  |
  = help: the following other types implement trait `serde::de::Deserialize<'de>`:
            &'a [u8]
            &'a serde_json::raw::RawValue
            &'a std::path::Path
            &'a str
            ()
            (T0, T1)
            (T0, T1, T2)
            (T0, T1, T2, T3)
          and 129 others
  = note: required for `MyPath` to implement `serde::de::DeserializeOwned`
  = note: required for `axum::extract::Path<MyPath>` to implement `FromRequestParts<S>`
  = note: this error originates in the derive macro `TypedPath` (in Nightly builds, run with -Z macro-backtrace for more info)
